lib/sidekiq_unique_jobs/client/middleware.rb

```diff
@@ -63,7 +63,7 @@ def old_unique_for?
 
       def new_unique_for?
         connection do |conn|
-          return conn.set(payload_hash, item['jid'], nx: true, ex: expires_at)
+          return conn.set(payload_hash, item['jid'], nx: true, ex: expires_at) || conn.get(payload_hash) == item['jid']
```

Comment:
@pik It sounds like we have different use cases. In our environment we don't want a duplicate job to be able to run now if it's scheduled in the future even if it's several hours out.